### PR TITLE
Manipuliation transition component activation

### PR DESCRIPTION
**Pull request by NerijusKondrotas**
Add manipulation ComponentActivationTransition

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] @maximilianpilz 
  * [x] @antipovi 
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] @maximilianpilz 
  * [x] @antipovi

### CODE REVIEW

File: src/t2iapi/activation_state/activation_state_requests.proto [hunk 1]
```diff
@@ -34,6 +34,19 @@ message SetAlertActivationRequest {
     AlertActivation activation = 2;  // value to set as the ActivationState value
 }
 
+/*
+Request to set the ActivationState of the component or metric with the given handle and the start and
+end @ActivationState value.
+ */
+message SetTransitionComponentActivationRequest {
```
Comment (antipovi) [inline src/t2iapi/activation_state/activation_state_requests.proto:41]: Set and Transition doesnt make sense imo
i think "Set" can be dropped from manipulation name, also from the rpc call

File: src/t2iapi/activation_state/activation_state_requests.proto [hunk 2]
```diff
@@ -34,6 +34,19 @@ message SetAlertActivationRequest {
     AlertActivation activation = 2;  // value to set as the ActivationState value
 }
 
+/*
+Request to set the ActivationState of the component or metric with the given handle and the start and
```
Comment (antipovi) [inline src/t2iapi/activation_state/activation_state_requests.proto:38]: misleading description, the objective of the manipulation is not very clear, write the expectancy
transition from one state to another is not mentioned

**Review by antipovi — CHANGES_REQUESTED** [review]

File: src/t2iapi/activation_state/activation_state_requests.proto [hunk 3]
```diff
@@ -34,6 +34,17 @@ message SetAlertActivationRequest {
...
+end @ActivationState value.
+*/
+message TransitionComponentActivationRequest {
+    string handle = 1;  // descriptor handle of the descriptor for which the ActivationState shall be set
```
Comment (antipovi) [inline src/t2iapi/activation_state/activation_state_requests.proto:42]: @ActivationState

File: src/t2iapi/activation_state/service.proto [hunk 4]
```diff
@@ -34,6 +34,12 @@ service ActivationStateService {
   rpc SetComponentActivation (t2iapi.activation_state.SetComponentActivationRequest)
       returns (BasicResponse);
 
+  /*
+  The ActivationState transition from the start to the end of the DeviceComponent or Metric with the given handle.
```
Comment (antipovi) [inline src/t2iapi/activation_state/service.proto:38]: @ActivationState

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - manipulation SetActivationStateAndUserConfirmableValue for metrics
 - stub files which are integrated into the python package
 - description for allowed combinations of InstanceIdentifier/@Root and InstanceIdentifier/@Extension
+- manipulation TransitionComponentActivation for metrics
```
Comment (antipovi) [inline CHANGELOG.md:15]: the manipulation is not only for metrics

**Review by antipovi — CHANGES_REQUESTED** [review]

**Comment by maximilianpilz** [comment]
replace

> // Describe the purpose of the pull request here and remove this line. Keep the checklist.

in the initial comment

**Review by maximilianpilz — CHANGES_REQUESTED** [review]
replace

> // Describe the purpose of the pull request here and remove this line. Keep the checklist.

in the initial comment

**Review by antipovi — APPROVED** [review]

**Review by maximilianpilz — APPROVED** [review]

**Pull request merged**